frame/two-phase-election-provider/src/benchmarking.rs

```diff
@@ -0,0 +1,358 @@
+	} verify {
+		assert_eq!(T::Currency::free_balance(&receiver), initial_balance - 10u32.into());
+		assert_eq!(T::Currency::reserved_balance(&receiver), 0u32.into());
+	}
+
+	#[extra]
+	create_snapshot {
+		assert!(<TwoPhase<T>>::snapshot().is_none());
+	}: {
+		<TwoPhase::<T>>::create_snapshot()
+	} verify {
+		assert!(<TwoPhase<T>>::snapshot().is_some());
+	}
+
+	submit {
+		let c in 1 .. (T::MaxSignedSubmissions::get() - 1);
+
+		// the solution will be worse than all of them meaning the score need to be checked against all.
+		let solution = RawSolution { score: [(1000_0000u128 - 1).into(), 0, 0], ..Default::default() };
+
+		<CurrentPhase<T>>::put(Phase::Signed);
+		<Round<T>>::put(1);
+
+		for i in 0..c {
+			<SignedSubmissions<T>>::mutate(|queue| {
+				let solution = RawSolution { score: [(1000_0000 + i).into(), 0, 0], ..Default::default() };
+				let signed_submission = SignedSubmission { solution, ..Default::default() };
+				// note: this is quite tricky: we know that the queue will stay sorted here. The
+				// last will be best.
+				queue.push(signed_submission);
+			})
+		}
+
+		let caller = frame_benchmarking::whitelisted_caller();
+		T::Currency::make_free_balance_be(&caller,  T::Currency::minimum_balance() * 10u32.into());
+
+	}: _(RawOrigin::Signed(caller), solution, c)
+	verify {
+		assert!(<TwoPhase<T>>::signed_submissions().len() as u32 == c + 1);
+	}
+
+	submit_unsigned {
+		// number of votes in snapshot.
+		let v in 2000 .. 3000;
+		// number of targets in snapshot.
+		let t in 500 .. 800;
+		// number of assignments, i.e. compact.len(). This means the active nominators, thus must be
+		// a subset of `v` component.
+		let a in 500 .. 1500;
+		// number of desired targets. Must be a subset of `t` component.
+		let d in 200 .. 400;
+
+		let witness = SolutionSize { voters: v, targets: t };
+		let raw_solution = solution_with_size::<T>(witness, a, d);
+
+		assert!(<TwoPhase<T>>::queued_solution().is_none());
+		<CurrentPhase<T>>::put(Phase::Unsigned((true, 1u32.into())));
+	}: _(RawOrigin::None, raw_solution, witness)
+	verify {
+		assert!(<TwoPhase<T>>::queued_solution().is_some());
+	}
+
+	// This is checking a valid solution. The worse case is indeed a valid solution.
+	feasibility_check {
+		// number of votes in snapshot.
+		let v in 2000 .. 3000;
+		// number of targets in snapshot.
+		let t in 500 .. 800;
+		// number of assignments, i.e. compact.len(). This means the active nominators, thus must be
+		// a subset of `v` component.
+		let a in 500 .. 1500;
+		// number of desired targets. Must be a subset of `t` component.
+		let d in 200 .. 400;
+
+		let size = SolutionSize { voters: v, targets: t };
+		let raw_solution = solution_with_size::<T>(size, a, d);
+
+		assert_eq!(raw_solution.compact.voter_count() as u32, a);
+		assert_eq!(raw_solution.compact.unique_targets().len() as u32, d);
+	}: {
+		assert_ok!(<TwoPhase<T>>::feasibility_check(raw_solution, ElectionCompute::Unsigned));
+	}
```

Comment:
Nothing to verify?

Reply:
No state changes happen there, so just `assert_ok`.